go14: add go 1.4.2

skatsuta opened the pull request:
Added Go 1.4.2.

dunn commented:
Let's add a conflict:
```ruby
conflicts_with "go", :because => "both install a `go` executable"
```

It's that or make this formula keg-only.

DomT4 commented on go14.rb:
```diff
@@ -0,0 +1,96 @@
+  version "1.4.2"
+  sha256 "299a6fd8f8adfdce15bc06bde926e7b252ae8e24dd5b16b7d8791ed79e7b5e9b"
+
+  bottle do
```
Cut this block. We'll add it on pull.

DomT4 commented on go14.rb:
```diff
@@ -0,0 +1,96 @@
+  option "without-godoc", "godoc will not be installed for you"
+  option "without-vet", "vet will not be installed for you"
+
+  deprecated_option "cross-compile-all" => "with-cc-all"
```
No such thing as a deprecated option in a new formula.

DomT4 commented on go14.rb:
```diff
@@ -0,0 +1,96 @@
+  option "with-cc-all", "Build with cross-compilers and runtime support for all supported platforms"
+  option "with-cc-common", "Build with cross-compilers and runtime support for darwin, linux and windows"
+  option "without-cgo", "Build without cgo"
+  option "without-godoc", "godoc will not be installed for you"
```
You either need to remove these two options or include the original:

```ruby
    if build.with?("godoc") || build.with?("vet")
      ENV.prepend_path "PATH", bin
      ENV["GOPATH"] = buildpath
      (buildpath/"src/golang.org/x/tools").install resource("gotools")

      if build.with? "godoc"
        cd "src/golang.org/x/tools/cmd/godoc/" do
          system "go", "build"
          (libexec/"bin").install "godoc"
        end
        bin.install_symlink libexec/"bin/godoc"
      end

      if build.with? "vet"
        cd "src/golang.org/x/tools/cmd/vet/" do
          system "go", "build"
          # This is where Go puts vet natively; not in the bin.
          (libexec/"pkg/tool/darwin_amd64/").install "vet"
        end
      end
    end
```

block.

skatsuta replied:
@DomT4 
I added the block, and tests at the end of the code.

skatsuta commented:
@dunn Thank you for your review!

- added the `conflict_with` statement
- cut `bottle` block
- removed the `deprecated_option` statements
- added `godoc` and `vet` blocks (build and test)

DomT4 commented:
Oh. Mike hopped over you today with his Go side-by-side configuration changes in https://github.com/Homebrew/homebrew-versions/commit/0761fbc415db69f97f4fffc291954ea4977540f4. Apologies for that. Thank you for wanting to contribute to Homebrew, we appreciate the submission :heart:
